src/sentry/deletions/defaults/group.py

```diff
@@ -159,6 +164,53 @@ def delete_events_from_snuba(self) -> None:
             eventstream.backend.end_delete_groups(eventstream_state)
 
 
+class IssuePlatformEventsDeletionTask(EventsBaseDeletionTask):
+    """
+    This class helps delete Issue Platform events which use the new Clickhouse light deletes.
+    """
+
+    dataset = Dataset.IssuePlatform
+
+    def chunk(self) -> bool:
+        """This method is called to delete chunks of data. It returns a boolean to say
+        if the deletion has completed and if it needs to be called again."""
+        events = self.get_unfetched_events()
+        if events:
+            self.delete_events_from_nodestore(events)
+            # This value will be used in the next call to chunk
+            self.last_event = events[-1]
+            # As long as it returns True the task will keep iterating
+            return True
+        else:
+            # Now that all events have been deleted from the eventstore, we can delete the events from snuba
+            self.delete_events_from_snuba()
+            return False
+
+    def delete_events_from_nodestore(self, events: Sequence[Event]) -> None:
+        # We delete by the occurrence_id instead of the event_id
+        node_ids = [
+            Event.generate_node_id(event.project_id, event._snuba_data["occurrence_id"])
+            for event in events
+        ]
+        nodestore.backend.delete_multi(node_ids)
+
+    def delete_events_from_snuba(self) -> None:
+        requests = []
+        for project_id, group_ids in self.project_groups.items():
+            query = DeleteQuery(
+                self.dataset.value,
+                column_conditions={"project_id": [project_id], "group_id": group_ids},
+            )
+            request = Request(
+                dataset=self.dataset.value,
+                app_id=self.referrer,
+                query=query,
+                tenant_ids=self.tenant_ids,
+            )
+            requests.append(request)
+        bulk_snuba_queries(requests)
```

Comment:
Cool, so we can actually submit deletion queries to snuba now, rather than relying on replacements?

Do these deletions run in real time, or are the enqueued?

Reply:
The rows immediately become unavailable to be queried, however, they will truly be deleted later.

Reply:
How long does this query take? Is it dependent on the number of occurrences/events associated with the group?

Reply:
Discussed this on slack. The deletion happens sync, so the query might time out for very large groups. We shouldn't release this on saas without handling cases like this, but I'm fine to merge for testing